app/controllers/api/base_controller/authentication.rb

```diff
@@ -12,22 +12,26 @@ def require_api_user_or_token
         elsif request.headers[HttpHeaders::AUTH_TOKEN]
           authenticate_with_user_token(request.headers[HttpHeaders::AUTH_TOKEN])
         else
-          authenticate_options = {
-            :require_user => true,
-            :timeout      => ::Settings.api.authentication_timeout.to_i_with_method
-          }
-
-          if (user = authenticate_with_http_basic { |u, p| User.authenticate(u, p, request, authenticate_options) })
-            auth_user_obj = userid_to_userobj(user.userid)
-            authorize_user_group(auth_user_obj)
-            validate_user_identity(auth_user_obj)
-            User.current_user = auth_user_obj
-          else
-            request_http_basic_authentication
-            return
-          end
+          authenticate_with_http_basic do |u, p|
+            begin
+              user = User.authenticate(
+                u, p, request,
+                :require_user => true,
+                :timeout      => ::Settings.api.authentication_timeout.to_i_with_method
+              )
+              auth_user_obj = userid_to_userobj(user.userid)
+              authorize_user_group(auth_user_obj)
+              validate_user_identity(auth_user_obj)
+              User.current_user = auth_user_obj
+            rescue MiqException::MiqEVMLoginError => e
+              raise AuthenticationError, e.message
+            end
+          end or raise AuthenticationError
```

Comment:
why the `or`?  that's generally a no-no

Reply:
That's just a Perl-ism - `|| raise` isn't valid Ruby, so the alternative is to make this (perhaps) less readable by assigning to a variable that's only used to check that this is non-nil

Reply:
honestly, I find the variable more readable...or use tap (I do find tap less readable).

See also https://github.com/bbatsov/ruby-style-guide#no-and-or-or

`||` is valid Ruby, so not sure why you think it's not...maybe you are using it without parenthesis?

```ruby
[1] pry(main)> false || raise("blah")
RuntimeError: blah
from (pry):20:in `<main>'
[2] pry(main)> false || raise "blah"
SyntaxError: unexpected tSTRING_BEG, expecting keyword_do or '{' or '('
false || raise "blah"
                ^
```

Reply:
Ah yes, I must have misinterpreted that error, but I find having to know that that `raise` requires parentheses perhaps a bit arcane. I'll change to use the variable.

FWIW I prefer Avdi's take on the no `or`/`and` rule :wink: